overlord/hookstate/hookmgr.go

```diff
@@ -217,6 +219,21 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 	return nil
 }
 
+func runHookImpl(c *Context, tomb *tomb.Tomb) ([]byte, error) {
+	return runHookAndWait(c.SnapName(), c.SnapRevision(), c.HookName(), c.ID(), tomb)
+}
+
+var runHook = runHookImpl
+
+// MockRunHook mocks the actual invocation of hooks for tests.
+func MockRunHook(hookInvoke func(c *Context, tomb *tomb.Tomb) ([]byte, error)) (restore func()) {
```

Comment:
yes, MockCommand would need to setup most of the daemon stuff which we don't need to test here and expect in other scenarios